### PR TITLE
Update Cilium related docs

### PR DESCRIPTION
**Pull request by aanm**
Cilium v1.6 has been released for a while and the documentation available for users is deprecated an error prone causing some friction to when running Cilium with Minikube and Kubeadm, this PR updates Cilium related docs to include the following changes:

* New location of cilium.yaml file.
* By default, Cilium no longer depends on an external etcd store.
* New minikube min version requirement.
* Cilium can work on top of other CNI plugins.
* Cilium can run without kube-proxy.

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/network-policy-provider/cilium-network-policy.md [hunk 1]
```diff
@@ -86,14 +95,8 @@ cilium-6rxbd    1/1     Running   0          1m
 ...
 ```
 
-There are two main components to be aware of:
-
-- One `cilium` Pod runs on each node in your cluster and enforces network policy
+`cilium` Pod runs on each node in your cluster and enforces network policy
```
Comment (sftim) [inline content/en/docs/tasks/administer-cluster/network-policy-provider/cilium-network-policy.md]: nit:
```suggestion
A `cilium` Pod runs on each node in your cluster and enforces network policy
```

Comment (aanm) [reply content/en/docs/tasks/administer-cluster/network-policy-provider/cilium-network-policy.md]: Thanks for the review @sftim

**Comment by sftim** [comment]
This looks ready for tech review

File: content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md [hunk 2]
```diff
@@ -319,23 +319,17 @@ kubectl apply -f https://docs.projectcalico.org/v3.8/manifests/canal.yaml
...
-_Note_: If you are running kubeadm in a single node please untaint it so that
-etcd-operator pods can be scheduled in the control-plane node.
-
-```shell
-kubectl taint nodes <node-name> node-role.kubernetes.io/master:NoSchedule-
-```
+Cilium can provide kube-proxy functionality. To disable kube-proxy, pass
+`--skip-phases=addon/kube-proxy` to `kubeadm init`. Afterwards, you will need
+to configure Cilium to enable the BPF NodePort feature and to specify the kube-apiserver
+address, see [Kubernetes without kube-proxy](https://docs.cilium.io/en/stable/gettingstarted/kubeproxy-free/).
```
Comment (neolit123) [inline content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md]: i would change this whole paragraph to the following:

> Cilium can be used as a replacement for kube-proxy, see [Kubernetes without kube-proxy](https://docs.cilium.io/en/stable/gettingstarted/kubeproxy-free/).

and move it right above:

> For more information about using Cilium with Kubernetes, see [Kubernetes Install guide for Cilium](https://docs.cilium.io/en/stable/kubernetes/).

my rationale is that we don't want to first class this yet:
- it's Beta as per the Cilium docs.
- i haven't heard comments from SIG Network about it.

Comment (aanm) [reply content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md]: @neolit123 Looks good to me as well. The [Kubernetes without kube-proxy](https://docs.cilium.io/en/stable/gettingstarted/kubeproxy-free/) already contains the information to how run kubeadm without kube-proxy so there's no point in duplicating it here.

Comment (aanm) [reply content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md]: Thanks for the review!

**Comment by neolit123** [comment]
/lgtm
on the `content/en/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm.md `
changes. the rest also look good, but are on sig-network.

consider merging on lazy consensus, if nobody from sig-network comes by.

**Pull request merged**